main.py

```diff
@@ -73,7 +73,10 @@ def read_files(args):
         )
         # Add collector into collectors set in data class
         data.collectors.add(collector)
-    return data
+
+    collect_lines=collect_file.readlines()

```

Comment:
relationlarda data'da olsun çünkü bunları daha sonradan diğer databaseler için de okumamız gerekecek.
Relationlar için de dataset.py da bir class oluştur. Bu kısmı senin hayal gücüne bırakıyorum.